reconcile/ocm/types.py

```diff
@@ -96,7 +96,7 @@ class Config:
 
 class OCMSpec(BaseModel):
     path: Optional[str]
-    spec: Union[OSDClusterSpec, ROSAClusterSpec]
+    spec: Union[OSDClusterSpec, ROSAClusterSpec, OCMClusterSpec]
```

Comment:
I think it might be good to add `Extra.forbid` to the `class Config` https://pydantic-docs.helpmanual.io/usage/model_config/

When passing raw dicts to that class like `OCMSpec(**data)`, then pydantic iterates over the types in order of declaration inside the union and the first fitting wins. I.e., we should try to always have the most specific type first (in this case, it is OSDClusterSpec, because it has more attributes). On top, `extra.Forbid` will make the matching more reliable, as only types which declare ALL given keys are considered.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/2906/commits/9df32e0e83de564750922dadd923238848647163

Reply:
adding it caused failures, and this is likely outside the scope of this PR